src/test/java/acceptance/queue/QueueSteps.java

```diff
@@ -188,22 +214,28 @@ public void go_live(List<ProcessingRuleRepresentation> listOfRules) {
         );
         queueBasedImplementationRunner = queueBasedImplementationRunnerBuilder.create();
 
+        queueBasedImplementationRunner.start();
         long timestampBefore = System.nanoTime();
         queueBasedImplementationRunner.run();
         long timestampAfter = System.nanoTime();
         processingTimeMillis = (timestampAfter - timestampBefore) / 1000000;
+        consumedMessagesCount = queueBasedImplementationRunner.getConsumedMessagesCount();
```

Comment:
The assertions should be done on the queues. We need to treat the production code as a black box.

Reply:
This came about due to a catch-22 situation, if you read the messages they are off the queue, which means you have consumed it whether it was successfully consumed or not. There can be a couple (or more) edge cases but they all end up being harder to resolve if the message leaves the queue. Hence I do a hack to satisfy the use case but now we are back at it to fix it, I wonder if there is a fix without maintaining our own mirror of the messages which I was trying to do (maybe badly).

Any thoughts on how to go about it? 

I know this as I have justed reverted all scaffolding code to help fix this issue and the queues are always empty or have incorrect number of messages in them.

Reply:
The other part of the issue, is the testing semantics - maybe we are testing for the queue contents at an incorrect lifecycle for SQS, these JMS tests might need adapting? 

JMS queues behave differently than SQS ones.